engine/core/source/org/pentaho/reporting/engine/classic/core/modules/misc/datafactory/sql/SimpleSQLReportDataFactory.java

```diff
@@ -251,85 +251,107 @@ protected TableModel parametrizeAndQuery( final DataRow parameters, final String
       final String[] preparedParameterNames ) throws SQLException {
     final boolean callableStatementQuery = isCallableStatementQuery( translatedQuery );
     final boolean callableStatementUsed = callableStatementQuery || isCallableStatement( translatedQuery );
-    final Statement statement;
+    final Statement statement = initStatement( parameters, translatedQuery, preparedParameterNames, callableStatementUsed );
+    try {
+      final Object queryLimit = parameters.get( DataFactory.QUERY_LIMIT );
+      try {
+        if ( queryLimit instanceof Number ) {
+          final Number i = (Number) queryLimit;
+          final int max = i.intValue();
+          if ( max > 0 ) {
+            statement.setMaxRows( max );
+          }
+        }
+      } catch ( SQLException sqle ) {
+      // this fails for MySQL as their driver is buggy. We will not add workarounds here, as
+      // all drivers are buggy and this is a race we cannot win. Put pressure on the driver
+      // manufacturer instead.
+        logger.warn( "Driver indicated error: Failed to set query-limit: " + queryLimit, sqle );
+      }
+      final Object queryTimeout = parameters.get( DataFactory.QUERY_TIMEOUT );
+      try {
+        if ( queryTimeout instanceof Number ) {
+          final Number i = (Number) queryTimeout;
+          final int seconds = i.intValue();
+          if ( seconds > 0 ) {
+            statement.setQueryTimeout( seconds );
+          }
+        }
+      } catch ( SQLException sqle ) {
+        logger.warn( "Driver indicated error: Failed to set query-timeout: " + queryTimeout, sqle );
+      }
+
+      // Track the currently running statement - just in case someone needs to cancel it
+
+      final ResultSet res = getRS( translatedQuery, preparedParameterNames, statement );
+      final RowSetFactory aFactory = RowSetProvider.newFactory();
```

Comment:
There is no point in having this meaningless variable. Merge it with the next line. If anything goes wrong in the call chain, it will throw an proper exception.